gnovm/stdlibs/math/overflow/overflow.gno

```diff
@@ -2,8 +2,10 @@
 // NOTE: there was a bug with the original Quotient* functions, and
 // testing method. These have been fixed here, and tests ported to
 // tests/files/maths_int*.go respectively.
-// TODO: make PR upstream.
```

Comment:
It's funny because there's a mention of a change, but this change is not on `gnolang/overflow` :shrug: (It has no commits other than those already in the upstream)

I did some digging, this is the diff: https://www.diffchecker.com/gGNCxLdJ/ - red is library source, right is this file.

Aside from renaming Quotient -> Quo and fixing a typo, the changes seem to be mostly about turning this general Quotient function:

```go
// Quotient16 performs + operation on two int16 operands
// returning a quotient, a remainder and status
func Quotient16(a, b int16) (int16, int16, bool) {
	if b == 0 {
		return 0, 0, false
	}
	c := a / b
	status := (c < 0) == ((a < 0) != (b < 0))
	return c, a % b, status
}
```

To this:

```go
// Quo16 performs + operation on two int16 operands
// returning a quotient, a remainder and status
func Quo16(a, b int16) (int16, int16, bool) {
	if b == 0 {
		return 0, 0, false
	} else if b == -1 && a == math.MinInt16 {
		return 0, 0, false
	}
	c := a / b
	return c, a % b, true
}
```

(funny how the documentation says this is the `+` operation while this is clearly NOT the case).

It's also unclear to me how the `status` was removed -- it seems to me to be a reasonable overflow check to keep.

I'd say we go ahead with the move. I'll make an issue to fix up gnolang/overflow, and then we can port over the changes here.

Reply:
This update is actually fine. 

The one situation where division can cause an integer overflow in golang-style division is the "else-if" condition block. For example, for int8 ranging between -128 and 127, it would be -128 / - 1 == 128

The remainder can never cause an overflow based on the way % works in golang.

Therefore, if we don't have a div-by-zero condition and don't trigger this specific overflow condition, the "status" return value is always true.

Reply:
https://go.dev/play/p/nwmGZ_csqzt